docrepr/sphinxify.py

```diff
@@ -218,6 +225,68 @@ def generate_extensions(render_math):
     return extensions
 
 
+def wrap_docstring(oinfo):
+    """Generate a docstring wrapped in the appropriate Sphinx domain."""
+    should_wrap_docstring = bool(
+        oinfo['docstring']
+        and is_sphinx_markup(oinfo['docstring'])
+        and (oinfo['docstring'] != '<no docstring>')
+        and oinfo.get('type_name')
+        and oinfo.get('name')
+        )
+    if not should_wrap_docstring:
+        return oinfo['docstring']
+
+    if oinfo.get('definition'):
+        formatted_definition = oinfo.get('definition')
+    elif oinfo.get('init_definition'):
+        formatted_definition = oinfo.get('init_definition')
+    else:
+        formatted_definition = ''
+    if formatted_definition:
+        formatted_definition = re.sub(r'\(\n\s*', '(', formatted_definition)
+        formatted_definition = re.sub(r',\n\s*\)', ')', formatted_definition)
+        formatted_definition = re.sub(r',\n\s*', ', ', formatted_definition)
+        formatted_definition = formatted_definition.replace('\n', ' ')
+
+    indented_docstring = textwrap.indent(oinfo['docstring'], ' ' * 3)
+
+    type_name = oinfo['type_name']
+    if type_name not in [
+            'function', 'method', 'property', 'attribute', 'module']:
+        type_name = 'class'
+
+    wrapped_docstring = DOCSTRING_TEMPLATE.format(
+        kind=type_name,
+        name=oinfo['name'],
+        definition=formatted_definition,
+        docstring=indented_docstring,
+        )
+    return wrapped_docstring
+
+
+def wrap_class_docstring(oinfo):
+    """Wrap a class docstring wrapped in a Sphinx domain."""
+    should_wrap_docstring = bool(
+        oinfo.get('class_docstring')
+        and is_sphinx_markup(oinfo['class_docstring'])
+        and (oinfo['class_docstring'] != '<no docstring>')
+        and oinfo.get('type_name')
+        )
+    if not should_wrap_docstring:
+        return oinfo.get('class_docstring', '')
+
+    indented_docstring = textwrap.indent(oinfo['class_docstring'], ' ' * 3)
+
+    wrapped_docstring = DOCSTRING_TEMPLATE.format(
+        kind='class',
+        name=oinfo['type_name'],
+        definition='',
+        docstring=indented_docstring,
+        )
+    return wrapped_docstring
```

Comment:
I wonder if there would be a way to reuse `wrap_docstring` in `wrap_class_docstring`? 

Maybe it's not worth trying to do it, the code is quite simple.

Reply:
Yeah; I thought about it before but ended up just pushing it as it was since I was kinda tired, and it required some thinking and substantial refactoring. I went ahead and did so now, thanks!